cudaaligner/src/aligner_global_myers_banded.cpp

```diff
@@ -29,43 +33,312 @@ namespace genomeworks
 namespace cudaaligner
 {
 
-struct AlignerGlobalMyersBanded::Workspace
+namespace
 {
-    Workspace(int32_t max_alignments, int32_t max_n_words, int32_t max_target_length, DefaultDeviceAllocator allocator, cudaStream_t stream)
-        : pvs(max_alignments, max_n_words * (max_target_length + 1), allocator, stream)
-        , mvs(max_alignments, max_n_words * (max_target_length + 1), allocator, stream)
-        , scores(max_alignments, max_n_words * (max_target_length + 1), allocator, stream)
-        , query_patterns(max_alignments, max_n_words * 4, allocator, stream)
+
+constexpr int32_t n_alignments_initial_parameter = 1000; // for initial allocation - buffers will grow automatically if needed.
+
+using myers::WordType;
+constexpr int32_t word_size = sizeof(myers::WordType) * CHAR_BIT;
+
+template <typename T>
+using pinned_host_vector = std::vector<T, thrust::system::cuda::experimental::pinned_allocator<T>>;
```

Comment:
Should this be moved to `common/base/include/claraparabricks/genomeworks/types.hpp`?
Probably not necessary to do it right away

Reply:
Good point. However, I think we should not move it to types.hpp, but a separate header file. It implies including header files for a pinned allocator from thrust and std::vector, which I would prefer not to force on every compilation unit. How about `claraparabricks/genomworks/utils/pinned_host_vector.hpp`?

Reply:
Yes, that sounds good. If it's too much work you can leave it as it is for now as well.

Reply:
Ok, now I see you already changed it